### DISCUSSION TITLE
docker-compose에서 .env 환경 변수 설정 에러

### DISCUSSION THREAD

**Discussion by kwongiyeon** [100-hours-a-week/discussions#83] [개발]
도커 컴포즈 파일을 하나 만들어서 블루,그린으로 컨테이너 2개 실행 시키는 deploy.sh를 작성했습니다. 그리고 deploy.sh에서 만들어지는 컨테이너를 하나의 EC2에서 실행 시켜 blue-green으로 구현하려고 합니다.
.ENV를 도커 컴포즈에서 불러오도록 환경 변수를 설정해서 터미널에서 EXPORT 해놓고, EC2에 도커 컴포즈 yml 파일도 생성하기도 했습니다.
.env 값은 액션 시크릿에 기입되어 있는 상태입니다.

그런데 깃헙 액션을 실행 시켰을 때 .env가 불러와지지 않아서 컨테이너를 실행하지 못하고 있습니다. EC2로 파일 생성 및 복사할 때 deploy.sh와 docker-compose.yml 경로를 전부 /home/ubuntu로 지정했음에도 불구하고 .env를 인식하지 못했는지 err: time="2024-09-09T12:46:35Z" level=warning msg="The \"DB_PASSWORD\" variable is not set. Defaulting to a blank string."와 같은 에러만 발생합니다.
도커 컴포즈로 블루-그린 배포에 관하여 다른 팀원들한테 도움 요청해보고 인터넷을 참고해봤으나 아직까지 근본적인 원인도 해답도 찾지 못했습니다. 이 방법으로 안 되면 다른 방법을 고려할 생각입니다

![image](https://github.com/user-attachments/assets/4506d71c-fc89-441a-a242-6556a9921498)

**Comment by ej31** [comment]
# 답변

- 우선 CD 측에서 (이 경우에선 Github Actions 겠지요?) 파일을 서버로 모두 넘겨주지 않은 것으로 보입니다.
  - 배포 서버측에서 .env 파일을 CD를 통해 정상적으로 받지 못했기 때문에 발생한 문제로 보입니다.
-  개발 환경 변수를 어떤 방식으로 넘긴 것인지 정확하게 진단해보기 위해 문제가 발생한 github workflow 파일 링크를 주시면 더 정확하게 답변 드릴 수 있을 것 같습니다. 💁‍♂️💁‍♂️
